openstack_controller/datadog_checks/openstack_controller/components/bare_metal.py

```diff
@@ -108,6 +111,24 @@ def _report_portgroups(self, config, tags, node_id):
                 self.check.log.debug("portgroup: %s", portgroup)
                 self.check.gauge(IRONIC_NODE_PORTGROUP_COUNT, 1, tags=tags + portgroup['tags'], hostname=item['uuid'])
 
+    @Component.register_global_metrics(ID)
+    @Component.http_error()
+    def _report_ports(self, config, tags):
+        report_ports = config.get('ports', True)
+        if report_ports:
+            data = self.check.api.get_baremetal_ports()
+            for item in data:
+                port = get_metrics_and_tags(
+                    item,
+                    tags=IRONIC_PORT_TAGS,
+                    prefix=IRONIC_PORT_PREFIX,
+                    metrics=[IRONIC_PORT_COUNT],
+                )
+                self.check.log.debug("port: %s", port)
+                self.check.gauge(IRONIC_PORT_COUNT, 1, tags=tags + port['tags'])
```

Comment:
How many times are we reporting `IRONIC_PORT_COUNT `?

Reply:
You're right, looping through the list of metrics is unnecessary when `IRONIC_PORT_COUNT` is the only metric. I have removed the loop.